File: php/features.php
<?php
declare(strict_types=1);

// ProxyChecker SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class ProxyCheckerFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new ProxyCheckerBaseFeature();
            case "test":
                return new ProxyCheckerTestFeature();
            default:
                return new ProxyCheckerBaseFeature();
        }
    }
}

File: py/features.py
# ProxyChecker SDK feature factory

from feature.base_feature import ProxyCheckerBaseFeature
from feature.test_feature import ProxyCheckerTestFeature


def _make_feature(name):
    features = {
        "base": lambda: ProxyCheckerBaseFeature(),
        "test": lambda: ProxyCheckerTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
